### PR TITLE
Convert tests to client code

### PR DESCRIPTION
**Pull request by melodeeli98**

### CODE REVIEW

**Review by alex-stanescu — APPROVED** [review]
Consider checking if the regex matches the args that would be cool

File: src/test/java/com/github/ermay12/regex/PhoneNumberClientCode.java [hunk 1]
```diff
@@ -26,5 +27,13 @@ public void testPhoneNumber() {
                                     capture(DIGIT),
```
Comment (ermay12) [inline src/test/java/com/github/ermay12/regex/PhoneNumberClientCode.java]: can you get rid of all the unused captures?

File: src/test/java/com/github/ermay12/regex/RomanNumeralClientCode.java [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+                ),
+                oneOf(
+                        concatenate(string("I"), union('X', 'V')),
+                        concatenate(optional("V"), repeat("I", 0, 3))
+                )
+        );
+        String input = "Roman numeral MCMXCIX is my favorite.";
+
+        Stream<RegexMatch> results = regex.getMatches(input);
+        results.forEach(result -> {
```
Comment (ermay12) [inline src/test/java/com/github/ermay12/regex/RomanNumeralClientCode.java]: this doesn't test.  Only prints

File: src/test/java/com/github/ermay12/regex/PasswordClientCode.java [hunk 3]
```diff
@@ -4,11 +4,10 @@
...
-import static org.junit.Assert.assertEquals;
 
-public class PasswordTest {
+public class PasswordClientCode {
     @Test
-    public void testPassword() {
+    public void validatePassword() {
         Regex regex = new Regex(
                 LINE_START,
                 lookahead(new Regex(anyAmount(WILDCARD), single(range('a', 'z')))),
```
Comment (ermay12) [inline src/test/java/com/github/ermay12/regex/PasswordClientCode.java]: single is redundant

**Review by ermay12 — CHANGES_REQUESTED** [review]

File: src/test/java/com/github/ermay12/regex/PhoneNumberClientCode.java (hunk 1 above)
Comment (alex-stanescu) [reply src/test/java/com/github/ermay12/regex/PhoneNumberClientCode.java]: They're not supposed to be redundant - you should be able to do something like `result.get().group(i)` to get the i'th digit in the phone number

Comment (alex-stanescu) [reply src/test/java/com/github/ermay12/regex/PhoneNumberClientCode.java]: so for instance you can do .group(1 - 3) to get the area code

**Pull request merged**